crates/router/src/connector/nexinets.rs

```diff
@@ -200,9 +204,15 @@ impl ConnectorIntegration<api::Authorize, types::PaymentsAuthorizeData, types::P
         &self,
         req: &types::PaymentsAuthorizeRouterData,
     ) -> CustomResult<Option<types::RequestBody>, errors::ConnectorError> {
-        let req_obj = nexinets::NexinetsPaymentsRequest::try_from(req)?;
+        let connector_router_data = nexinets::NexinetsRouterData::try_from((
+            &self.get_currency_unit(),
+            req.request.currency,
+            req.request.amount,
+            req,
+        ))?;
+        let connector_req = nexinets::NexinetsPaymentsRequest::try_from(&connector_router_data)?;
         let nexinets_req = types::RequestBody::log_and_get_request_body(
-            &req_obj,
+            &connector_req,
```

Comment:
Similarly you have to implement for `api::Capture`, `api::Void`, `api::execute`
the fields will be `req.request.amount_to_capture`, `req.request.amount`, `req.request.refund_amount`  respectively.